File: output/gpt_4o/avatar/algo_based_translation/java/python/codeforces_456_A.py

import sys

def main():
    n = int(sys.stdin.readline().strip())
    a = [0] * n
    b = [0] * n
    
    temp = 0
    q = 0
    w = 0
    e = 0
    f = False
    
    for i in range(n):
        a[i], b[i] = map(int, sys.stdin.readline().strip().split())
        
        if temp < a[i]:
            if q > b[i]:
                f = True
            q = b[i]
            temp = a[i]
        
        if temp > a[i]:
            if q < b[i]:
                f = True
            q = b[i]
            temp = a[i]
        
        if a[i] > w:
            w = a[i]
            if b[i] < e:
                f = True
            e = b[i]
        
        if a[i] < w:
            if b[i] > e:
                f = True

    if f:
        print("Happy Alex")
    else:
        print("Poor Alex")

if __name__ == "__main__":
    main()

File: output/gpt_4o/avatar/algo_based_translation/python/java/codeforces_456_A.java

import java.util.*;

public class codeforces_456_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = Integer.parseInt(scanner.nextLine());
        List<int[]> m = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            String[] input = scanner.nextLine().split(" ");
            int[] pair = new int[]{Integer.parseInt(input[0]), Integer.parseInt(input[1])};
            m.add(pair);
        }

        m.sort((a, b) -> Integer.compare(b[0], a[0]));

        int[] a = m.get(0);

        for (int j = 1; j < m.size(); j++) {
            int[] i = m.get(j);
            if (i[1] > a[1]) {
                System.out.println("Happy Alex");
                return;
            }
            a = i;
        }

        System.out.println("Poor Alex");
    }
}

